### PR TITLE
[quant] Add support for multiple inputs in fusion pattern

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73572

Summary:
Previously we can't specify how to get extra inputs for fused ops in backend_config_dict,
for example, for patterns like:
(torch.add, (nn.BatchNorm2d, nn.Conv2d), MatchAllNode)

where nn.Conv2d is the root node, the extra MatchAllNode (the input for original torch.add) would be lost
This PR added a "extra_inputs_getter" key in the backend_config_dict, which allows user to provide a function,
that can return a list of extra input node for the fused op given the matched node pattern. In this case,
we need a function that returns the node that matches with `MatchAllNode`, it would be something like the following:

```
def extra_inputs_getter(pattern):
    add, conv_bn, extra_input = pattern
    return [extra_input]
```

Test Plan:
python test/test_quantization.py TestFuseFx.test_fusion_pattern_with_multiple_inputs

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34553210](https://our.internmc.facebook.com/intern/diff/D34553210)

### CODE REVIEW

File: torch/ao/quantization/fx/backend_config/utils.py [hunk 1]
```diff
@@ -101,3 +101,22 @@ def get_root_node(node_pattern):
...
 
     return root_node_getter_mapping
+
+def get_fusion_pattern_to_extra_inputs_getter(
+        backend_config_dict: Dict[str, Any]) -> Dict[Pattern, Callable]:
+    """ Get a map from fusion pattern to a function that returns extra input nodes
+    from the fusion pattern, in the order required by the root node. This is optional,
+    if not specified, we will not copy over any extra inputs for the root node.
+    Example:
+    # given a pattern, return the input node that need to be copied to the root node
```
Comment (vkuzo) [inline torch/ao/quantization/fx/backend_config/utils.py:111]: can we add a more specific example?  It's hard to understand what this means from just the docblock

**Review by vkuzo — APPROVED** [review]

File: torch/ao/quantization/fx/backend_config/utils.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/fx/backend_config/utils.py:111]: yeah sure
